sqlglot/dialects/hive.py

```diff
@@ -631,3 +631,15 @@ def datatype_sql(self, expression: exp.DataType) -> str:
         def version_sql(self, expression: exp.Version) -> str:
             sql = super().version_sql(expression)
             return sql.replace("FOR ", "", 1)
+
+        def struct_sql(self, expression: exp.Struct) -> str:
+            values = []
+
+            for i, e in enumerate(expression.expressions):
+                if isinstance(e, exp.PropertyEQ):
+                    self.unsupported("Hive does not support named structs.")
```

Comment:
I know this is gonna be annoying, but FWIW Spark v2 supports `STRUCT(1 AS a)` (so maybe this is also supported by Hive?) and also I found an [SO thread](https://stackoverflow.com/a/32386011) that shows a different function `select named_struct("a", 1)`, supposedly supported by Hive as well AFAICT. Works in both Spark v2 and v3.

Reply:
no, that's a plugin